File: Алгоритмы/Лабораторная работа 25/Си/1/stack.c
#include "stack.h"
#include <stdio.h>
#include <stdlib.h>

//Добавление элемента в стек
void push(int stack[], int *len_stack, int number){
	stack[(*len_stack)++] = number;
}

//Удаление элемента из стека
int pop(int stack[], int *len_stack){
	if (*len_stack == 0)
		return(stderr);
	
	return stack[--(*len_stack)];
}

//Вывод стека
void printStack(int stack[], int len_stack){
	int i = 0;
	while (i != len_stack)
		printf("%d ", stack[i++]);
	printf("\n");
}

File: algoritm/stack.py
stack = []
n = 100
while n != 0:
    print('1. Добавить \n 2. Удалить')
    num = int(input())
    if num == 1:
        print('Введите число.')
        n = input()
        stack.append(n)
        print(stack)
    elif num == 2:
        print('Удалить элемент.')
        stack.pop()
        print(stack)
